File: app/src/main/java/de/metzgore/beansplan/util/di/components/MainActivitySubComponent.java
package de.metzgore.beansplan.util.di.components;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;
import de.metzgore.beansplan.MainActivity;

@Subcomponent
public interface MainActivitySubComponent extends AndroidInjector<MainActivity> {
    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<MainActivity> {
    }
}

File: app/src/main/java/de/metzgore/beansplan/util/di/components/MainActivitySubComponent.kt
package de.metzgore.beansplan.util.di.components

import dagger.Subcomponent
import dagger.android.AndroidInjector
import de.metzgore.beansplan.MainActivity

@Subcomponent
interface MainActivitySubComponent : AndroidInjector<MainActivity> {
    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<MainActivity>()
}
